src/protocol.cpp

```diff
@@ -238,7 +238,7 @@ CInv::CInv(const std::string& strType, const uint256& hashIn)
         }
     }
     if (i == ARRAYLEN(ppszTypeName))
-        LogPrint("net", "CInv::CInv(string, uint256) : unknown type '%s'\n", strType);
+        throw std::out_of_range(strprintf("CInv::CInv(string, uint256): unknown type '%s'", strType));
```

Comment:
I recommend always using braces after conditionals.  It makes it a little easier to see what code the conditional covers.  Omitting the braces has contributed to some well known security vulnerabilities, in particular in openssl.
Same applies below in GetCommand.

Nit++